tracer/src/Datadog.Trace/IAST/Dataflow/AspectAttribute.cs

```diff
@@ -0,0 +1,156 @@
+
+            parameters.Add(targetMethod.Quote());
+            parameters.Add(targetType.Quote());
+            parameters.Add(paramShift);
+            parameters.Add(boxParam);
+            parameters.Add(filters);
+            parameters.Add(aspectType);
+            parameters.Add(string.Join(",", (vulnerabilityTypes ?? new VulnerabilityType[0]).Select(i => i.ToString()).ToArray()).Quote());
+
+            var targetMethodMatch = nameSplitter.Match(targetMethod);
+            TargetMethodAssemblies = GetAssemblyList(targetMethodMatch.Groups[1].Value);
+            TargetMethod = targetMethodMatch.Groups[2].Value;
+            TargetMethodType = targetMethodMatch.Groups[3].Value;
+            TargetTypeAssemblies = TargetMethodAssemblies;
+            TargetType = TargetMethodType;
+
+            if (!string.IsNullOrEmpty(targetType))
+            {
+                var targetTypeMatch = nameSplitter.Match(targetType);
+                TargetTypeAssemblies = GetAssemblyList(targetTypeMatch.Groups[1].Value);
+                TargetType = targetTypeMatch.Groups[3].Value;
+            }
+
+            AspectType = aspectType;
+            VulnerabilityTypes = vulnerabilityTypes ?? new VulnerabilityType[0];
+            IsVirtual = (TargetMethodType != TargetType);
+
+            ParamShift = paramShift;
+            BoxParam = boxParam;
+            Filters = filters;
+        }
+
+        // Target method data (base virtual)
+        public List<string> TargetMethodAssemblies { get; private set; }
+
+        public string TargetMethodType { get; private set; }
+
+        public string TargetMethod { get; }
+
+        // Final type data
+        public List<string> TargetTypeAssemblies { get; private set; }
+
+        public string TargetType { get; private set; }
+
+        public bool IsVirtual { get; }
+
+        public int[] ParamShift { get; } // Number of parameters to move up in stack before inyecting the Aspect
+
+        public bool[] BoxParam { get; } // Box parameter before adding call
+
+        public AspectFilter[] Filters { get; } // Filters applied to aspect insertion
+
+        public AspectType AspectType { get; private set; }
+
+        public VulnerabilityType[] VulnerabilityTypes { get; private set; }
+
+        public override string ToString()
+        {
+            return string.Format("[{0}({1})]", GetType().Name.Replace("Attribute", string.Empty), string.Join(",", parameters.Select(i => ConvertToString(i)).ToArray()));
+        }
+
+        internal static List<string> GetAssemblyList(string expression)
+        {
+            if (!string.IsNullOrEmpty(expression))
+            {
+                return new List<string>();
+            }
+
+            return expression.Split(',').Select(e => e.Trim()).Where(e => !string.IsNullOrEmpty(e)).ToList<string>();
+        }
+
+        internal static string? ConvertToString(object? obj)
```

Comment:
The method always returns a not null string, the ? could be removed

Reply:
Fixed